rules/Php74/Rector/LNumber/AddLiteralSeparatorToNumberRector.php

```diff
@@ -113,7 +113,7 @@ public function refactor(Node $node): ?Node
         }
 
         if (\str_contains($numericValueAsString, '.')) {
-            [$mainPart, $decimalPart] = explode('.', $numericValueAsString);
+            [$mainPart, $decimalPart] = explode('.', (string) $numericValueAsString);
```

Comment:
it seems got `PHPStan\Type\IntersectionType` for it: 

```php
PHPStan\Type\IntersectionType #79746
   types: array (2)
   |  0 => PHPStan\Type\StringType #44887
   |  1 => PHPStan\Type\Accessory\AccessoryNumericStringType #44890
```

Reply:
resolved with `$this->nodeTypeAnalyzer->isStringyType($type)` check https://github.com/rectorphp/rector-src/pull/1655/commits/ed049e385c64e0512d1fe875352c1f3383845781